shm/Player.hpp

```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include <memory>
+
+#include "Cargo.hpp"
+#include "Ship.hpp"
+
+class Player {
+public:
+    Player(std::unique_ptr<Ship> ship, size_t money);
+
+    size_t getMoney() const { return money_; };
+    size_t getAvaliableSpace() const { return avaliableSpace_; };
+    size_t getSpeed() const { return ship_->getSpeed(); };
+    Cargo* getCargo(size_t index) { return ship_->getCargo(index); };
```

Comment:
You can check if ship exist and you are not try to read non existing value.

Reply:
You cannot create Player object without Ship so it's redundant

Reply:
So, what about situation when you pass index 5 to getCargo and you have only 2 element in vector<Cargo> ?

Reply:
It will be resolved by ship's method :)

Reply:
Ok  :)